src/stan-users-guide/odes.Rmd

```diff
@@ -257,124 +258,95 @@ Note that when the system is linear, it is a good idea to try
-The initial state `y`, the initial time `t0`, the requested solution
-times `ts`, or the parameters `theta` may involve parameters.
-
+compute the solution of the ODE at `T` output times and add simulated
+measurement noise using `normal_rng`.
 
 ### Estimating System Parameters and Initial State {-}
 
-Stan provides statistical inference for unknown initial states and/or
-parameters.  The ODE solver will be used deterministically to produce
-predictions, much like the linear predictor does in a generalized
-linear model.  These states will then be observed with measurement error.
-
+With noisy observations of the solution it is possible in Stan to estimate
+both parameters and initial conditions. As an example, this is a harmonic
+oscillator example where both the system parameter `theta` and initial state
+`y0` are parameters:
 
 ```
 functions {
-  real[] sho(real t,
-             real[] y,
-             real[] theta,
-             real[] x_r,
-             int[] x_i) {
-    real dydt[2];
+  vector sho(real t,
+             vector y,
+             real theta) {
+    vector[2] dydt;
     dydt[1] = y[2];
-    dydt[2] = -y[1] - theta[1] * y[2];
+    dydt[2] = -y[1] - theta * y[2];
     return dydt;
   }
 }
 data {
   int<lower=1> T;
-  real y[T,2];
+  vector[2] y[T];
   real t0;
   real ts[T];
 }
-transformed data {
-  real x_r[0];
-  int x_i[0];
-}
 parameters {
   real y0[2];
   vector<lower=0>[2] sigma;
-  real theta[1];
+  real theta;
 }
 model {
-  real y_hat[T,2];
-  sigma ~ cauchy(0, 2.5);
+  vector[2] y_hat[T];
+  sigma ~ normal(0, 2.5);
   theta ~ std_normal();
   y0 ~ std_normal();
-  y_hat = integrate_ode_rk45(sho, y0, t0, ts, theta, x_r, x_i);
+  y_hat = ode_rk45(sho, y0, t0, ts, theta);
   for (t in 1:T)
     y[t] ~ normal(y_hat[t], sigma);
 }
 ```
 <a name="id:sho-both.figure"></a>
 
-This Stan program allows estimates of unknown initial conditions `y0`
-and system parameter `theta` for the simple harmonic oscillator with
-independent normal measurement error.
-
-A Stan program that can be used to estimate both the initial state and
-parameter value for the simple harmonic oscillator given noisy
-observations is given above.  Compared to the program for simulation,
-the program to estimate parameters uses the `integrate_ode` function
+Compared to the program for simulation,
+the program to estimate parameters uses the `ode_rk45` function
 in the model block rather than the generated quantities block.  There
-are Cauchy priors on the measurement error scales `sigma` and standard
+are half-normal priors on the measurement error scales `sigma` and standard
 normal priors on the components of parameter array `theta` and initial
-state parameter array `y0`.  The solutions to the ODE are then
-assigned to an array `y_hat`, which is then used as the location in
+state parameter array `y0`.  The solutions to the ODE are
+assigned to `y_hat`, which is then used as the location in
```

Comment:
Is this example in the example repo as well? If so may be good to update there as well

Reply:
I checked and the footnotes stuff hasn't made it this far in the manual yet, so there's no updating to be done there.

Reply:
@SteveBronder was talking about the `example-models` repo.   

The first footnotes PR is still pending:  https://github.com/stan-dev/docs/pull/164